### PR TITLE
[Fizz] Optimize end tags chunks

### PR DESCRIPTION
**Pull request by yujunjung**
## Summary

By caching '</' + tag + '>' `Chunk`, my hacker news ssr benchmarks showed an improvement of over 10% in requests/second

Benchmark: https://github.com/yujunjung/optimize-end-tag-fizz
Requests/second increased from 815 to 910

### CODE REVIEW

**Comment by acdlite** [comment]
Interesting benchmark. I wonder if we should have a hardcoded switch statement for the most common tags: `div`, `p`, `span`, etc. And then fall back to a map. Could you try measuring that?

File: packages/react-dom-bindings/src/server/ReactFizzConfigDOM.js [hunk 1]
```diff
@@ -2509,7 +2509,7 @@ function pushStyleImpl(
     target.push(stringToChunk(escapeTextForBrowser('' + child)));
   }
   pushInnerHTML(target, innerHTML, children);
-  target.push(endTag1, stringToChunk('style'), endTag2);
+  target.push(endChunkForTag('style'));
```
Comment (acdlite) [inline packages/react-dom-bindings/src/server/ReactFizzConfigDOM.js:2512]: might as well precompute these

Comment (yujunjung) [reply packages/react-dom-bindings/src/server/ReactFizzConfigDOM.js:2512]: not well, it is already cached so the computation that causes the memory allocation is only executed once

Comment (yujunjung) [reply packages/react-dom-bindings/src/server/ReactFizzConfigDOM.js:2512]: @acdlite

**Comment by acdlite** [comment]
I think this looks good to merge but I'll let @sebmarkbage or @gnoff decide

**Review by gnoff — APPROVED** [review]
I wonder if it'd be impactful to just statically reference precomputed chunks for start and end tags for every function that pushes with an explicit type. So no new switch but everywhere we can avoid the map lookup we do and we only use the map for the generic path where the type is dynamic. Either way let's get this merged. LMK if you experiment with this other idea

**Pull request merged**